src/middlewares/restrict-user-details.js

```diff
@@ -0,0 +1,102 @@
+module.exports = (config, { strapi }) => {
+  return async (ctx, next) => {
+    await next();
+    try {
+      if (ctx.request.method === "GET" && ctx.request.url.includes("/api/")) {
```

Comment:
Are we checking `/api/` to filter out requests from admin dashboard?

Reply:
Correct